app/screens/more_dms/more_dms.tsx

```diff
@@ -0,0 +1,409 @@
+        },
+        noResultText: {
+            fontSize: 26,
+            color: changeOpacity(theme.centerChannelColor, 0.5),
+        },
+    };
+});
+
+export default function MoreDirectMessages({
+    componentId,
+    currentTeamId,
+    currentUserId,
+    restrictDirectMessage,
+    teammateNameDisplay,
+}: Props) {
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+    const intl = useIntl();
+    const {formatMessage} = intl;
+
+    const searchTimeoutId = useRef<NodeJS.Timeout | null>(null);
+    const next = useRef(true);
+    const page = useRef(-1);
+    const mounted = useRef(false);
+
+    const [profiles, setProfiles] = useState<UserProfile[]>([]);
+    const [searchResults, setSearchResults] = useState<UserProfile[]>([]);
+    const [loading, setLoading] = useState(false);
+    const [term, setTerm] = useState('');
+    const [startingConversation, setStartingConversation] = useState(false);
+    const [selectedIds, setSelectedIds] = useState<{[id: string]: UserProfile}>({});
+    const selectedCount = Object.keys(selectedIds).length;
+
+    const isSearch = Boolean(term);
+
+    const addToProfiles = useRef((newProfiles: UserProfile[]) => setProfiles([...profiles, ...newProfiles]));
+    useEffect(() => {
+        addToProfiles.current = (newProfiles: UserProfile[]) => setProfiles([...profiles, ...newProfiles]);
+    }, [profiles]);
+
+    const loadedProfiles = ({users}: {users?: UserProfile[]}) => {
+        if (mounted.current) {
+            if (users && !users.length) {
+                next.current = false;
+            }
+
+            page.current += 1;
+            setLoading(false);
+            if (users && users.length) {
+                addToProfiles.current(users);
+            }
+        }
+    };
+
+    const clearSearch = useCallback(() => {
+        setTerm('');
+        setSearchResults([]);
+    }, []);
+
+    const getProfiles = useCallback(debounce(() => {
+        if (next.current && !loading && !term && mounted.current) {
+            setLoading(true);
+            if (restrictDirectMessage) {
+                fetchProfilesInTeam(serverUrl, currentTeamId, page.current + 1, General.PROFILE_CHUNK_SIZE).then(loadedProfiles);
+            } else {
+                fetchProfiles(serverUrl, page.current + 1, General.PROFILE_CHUNK_SIZE).then(loadedProfiles);
+            }
+        }
+    }, 100), [loading, isSearch, restrictDirectMessage, serverUrl, currentTeamId]);
+
+    const handleRemoveProfile = useCallback((id: string) => {
+        const newSelectedIds = Object.assign({}, selectedIds);
+
+        Reflect.deleteProperty(newSelectedIds, id);
+
+        setSelectedIds(newSelectedIds);
+    }, [selectedIds]);
+
+    const createDirectChannel = useCallback(async (id: string): Promise<boolean> => {
+        const user = profiles.find((u) => u.id === id);
+
+        const displayName = displayUsername(user, intl.locale, teammateNameDisplay);
```

Comment:
interesting.. if you have the displayName here I wonder if it can be passed as an optional argument to `makeDirectChannel` and then avoid having to do the `fetchMissingSidebarInfo` call...

thoughts ?